ui/src/overview.tsx

```diff
@@ -359,83 +429,132 @@ function DemographicCharts({ cohort }: DemographicChartsProps) {
     }
 
     const demographicData = data.counts;
+    const chartConfigs =
+      underlay.uiConfiguration.demographicChartConfigs.chartConfigs;
+
     let totalCount = 0;
-    const demographicsByGender = new Map();
-    const demographicsByGenderAgeRace = new Map();
-    const demographicRaces = new Set();
+    const chartsData = chartConfigs.map(() => new Map());
+    const stackedProperties = chartConfigs.map(() => new Set());
 
     for (let i = 0; i < demographicData.length; i++) {
       const count = demographicData[i].count ?? 0;
-      const gender = demographicData[i].definition?.gender.stringVal ?? "Other";
-      const race = demographicData[i].definition?.race.stringVal ?? "Other";
-      const currentYear = new Date().getFullYear();
-      const yob = demographicData[i].definition?.year_of_birth?.int64Val ?? 0;
-      const age = parseInt(currentYear.toString()) - yob;
-      let ageRange = "";
-      if (age >= 65) {
-        ageRange = "65+";
-      } else if (age >= 45 && age <= 64) {
-        ageRange = "45-64";
-      } else if (age <= 44) {
-        ageRange = "18-44";
-      }
-
-      // Accumulate total count for results
       totalCount += count;
 
-      // Configure hash map to represent data for gender chart
-      if (!demographicsByGender.has(gender)) {
-        demographicsByGender.set(gender, 0);
-      }
-      demographicsByGender.set(
-        gender,
-        demographicsByGender.get(gender) + count
-      );
-
-      // Configure hash map to represent data for gender / age / race chart
-      const genderAgeRange = `${gender} ${ageRange}`;
-
-      if (!demographicsByGenderAgeRace.has(genderAgeRange)) {
-        demographicsByGenderAgeRace.set(genderAgeRange, new Map());
-      }
-
-      if (!demographicsByGenderAgeRace.get(genderAgeRange).has(race)) {
-        demographicsByGenderAgeRace.get(genderAgeRange).set(race, 0);
-      }
-
-      const currCountForRace = demographicsByGenderAgeRace
-        .get(genderAgeRange)
-        .get(race);
-
-      demographicsByGenderAgeRace
-        .get(genderAgeRange)
-        .set(race, currCountForRace + count);
+      chartConfigs.forEach((config, index) => {
+        // Determine the primary data groupings for the chart.
+        const primaryPropertyComponents: string[] = [];
+        config.primaryProperties.forEach((property) => {
+          // Temporarily check if property is age since API only supports year of birth.
+          const propertyName =
+            property.key === "age" ? "year_of_birth" : property.key;
+          const propertyValues = Object.entries(
+            demographicData[i]?.definition ?? {}
+          ).find((data) => data[0] === propertyName);
+
+          if (propertyValues) {
+            const dataIntegerVal = propertyValues[1].int64Val;
```

Comment:
This can be simplified as: `const value = propertyValues[1]?.int64Val ?? propertyValues[1]?.stringVal ?? propertyValues[1]?.boolVal`.

I think you'll be able to do the bucketing using `value` without checking the type because Javascript, but if not, use `typeof value === "number"`.